packages/test/test-end-to-end-tests/src/test/summaries.spec.ts

```diff
@@ -307,6 +324,74 @@ describeNoCompat("Summaries", (getTestObjectProvider) => {
 		);
 	});
 
+	it("should not call entry point before data store is requested", async () => {
+		const dataStoreFactory1 = new DataObjectFactory(
+			TestDataObjectType1,
+			TestDataObject1,
+			[],
+			[],
+		);
+		const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+			runtime.IFluidHandleContext.resolveHandle(request);
+		const registryStoreEntries = new Map<string, Promise<IFluidDataStoreFactory>>([
+			[dataStoreFactory1.type, Promise.resolve(dataStoreFactory1)],
+		]);
+		const containerRuntimeFactoryWithDefaultDataStore =
+			getContainerRuntimeApi(pkgVersion).ContainerRuntimeFactoryWithDefaultDataStore;
+		const runtimeFactory = new containerRuntimeFactoryWithDefaultDataStore(
+			dataStoreFactory1,
+			registryStoreEntries,
+			undefined,
+			[innerRequestHandler],
+		);
+		const loader = provider.createLoader([[provider.defaultCodeDetails, runtimeFactory]]);
+
+		// Create a container for the first client.
+		const container1 = await createAndAttachContainer(
+			provider.defaultCodeDetails,
+			loader,
+			provider.driver.createCreateNewRequest(provider.documentId),
+		);
+		assert.doesNotThrow(
+			async () => requestFluidObject<TestDataObject1>(container1, "/"),
+			"Initialization from container client should success.",
+		);
+
+		// Create a summarizer for the container and do a summary shouldn't throw.
+		const createSummarizerResult = await createSummarizerFromFactory(
+			provider,
+			container1,
+			dataStoreFactory1,
+			undefined,
+			containerRuntimeFactoryWithDefaultDataStore,
+			registryStoreEntries,
+		);
+		const summarizer = createSummarizerResult.summarizer;
+		assert.doesNotThrow(() => {
+			summarizer.summarizeOnDemand({ reason: "test" });
+		});
+
+		// In summarizer, load the data store should fail.
+		const summarizerContainer = await provider.loadContainer(runtimeFactory);
+		await assert.rejects(
+			async () => requestFluidObject<TestDataObject1>(summarizerContainer, "/"),
+
+			"This validates the summarizer client doesn't fully initialize the data object.",
+		);
+
+		// Load second container, load the data store again should success.
+		const url = (await container1.getAbsoluteUrl("/")) as string;
+		const container2 = await loader.resolve({ url });
+		assert.doesNotThrow(
```

Comment:
Why is this an assert.doesNotThrow whereas the validation above is await assert.rejects?
This should be an `await assert.rejects` as well because if you are not awaiting, its not really validating anything.

Reply:
I would expect this to fail as in the request would fail.

Reply:
Right. Change to expect error